src/Control/Algebra/NonDet/Church.hs

```diff
@@ -87,10 +84,10 @@ instance MonadTrans NonDetC where
   lift m = NonDetC (\ _ leaf _ -> m >>= leaf)
   {-# INLINE lift #-}
 
-instance (Algebra sig m, Effect sig) => Algebra (Empty :+: Choose :+: sig) (NonDetC m) where
-  alg (L Empty)          = empty
-  alg (R (L (Choose k))) = k True <|> k False
-  alg (R (R other))      = NonDetC $ \ fork leaf nil -> alg (handle (Leaf ()) (fmap join . traverse runNonDet) other) >>= fold fork leaf nil
+instance (Algebra sig m, Effect sig) => Algebra (NonDet :+: sig) (NonDetC m) where
+  alg (L (L Empty))      = empty
+  alg (L (R (Choose k))) = k True <|> k False
+  alg (R other)          = NonDetC $ \ fork leaf nil -> alg (handle (Leaf ()) (fmap join . traverse runNonDet) other) >>= fold fork leaf nil
```

Comment:
This totally works 🤘 as do uses of the effects wherever they occur in the sequence:

```haskell
λ run (runFail (runNonDet (runFresh (choose (pure 'a') (pure 'b'))))) :: Either String [Char]
Right "ab" 
```